react/features/base/conference/middleware.js

```diff
@@ -437,55 +435,28 @@ function _pinParticipant({ getState }, next, action) {
     const participants = state['features/base/participants'];
     const id = action.participant.id;
     const participantById = getParticipantById(participants, id);
-
-    if (typeof APP !== 'undefined') {
-        const pinnedParticipant = getPinnedParticipant(participants);
-        const actionName = id ? ACTION_PINNED : ACTION_UNPINNED;
-        const local
-            = (participantById && participantById.local)
-                || (!id && pinnedParticipant && pinnedParticipant.local);
-        let participantIdForEvent;
-
-        if (local) {
-            participantIdForEvent = local;
-        } else {
-            participantIdForEvent = actionName === ACTION_PINNED
-                ? id : pinnedParticipant && pinnedParticipant.id;
-        }
-
-        sendAnalytics(createPinnedEvent(
-            actionName,
-            participantIdForEvent,
-            {
-                local,
-                'participant_count': conference.getParticipantCount()
-            }));
-    }
-
-    // The following condition prevents signaling to pin local participant and
-    // shared videos. The logic is:
-    // - If we have an ID, we check if the participant identified by that ID is
-    //   local or a bot/fake participant (such as with shared video).
-    // - If we don't have an ID (i.e. no participant identified by an ID), we
-    //   check for local participant. If she's currently pinned, then this
-    //   action will unpin her and that's why we won't signal here too.
-    let pin;
-
-    if (participantById) {
-        pin = !participantById.local && !participantById.isFakeParticipant;
```

Comment:
Does this work with sharing youtube video as I see the logic about pining was checking isFakeParticipant?

Reply:
Yes. That logic only prevented the `conference.pinParticipant` function from being called in that case, but it's now never called anyway.

Reply:
Good point, thanks for taking a look!